scipy/special/tests/test_support_cupy_torch_jax.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from hypothesis import given, strategies
+import hypothesis.extra.numpy as npst
+from numpy.testing import assert_allclose
+
+from scipy.special._support_cupy_torch_jax import (get_array_special_func,
+                                                   array_special_func_map)
+from scipy.conftest import array_api_compatible
+from scipy import special
+from scipy._lib.array_api_compat.array_api_compat import numpy as np
+
+
+def test_dispatch_failure():
+    message = ("SciPy cannot dispatch to special function 'ndtr' "
+               "in array library 'object'.")
+    with pytest.raises(ValueError, match=message):
+        get_array_special_func('ndtr', xp=object)
+
+
+@array_api_compatible
+@given(data=strategies.data())
+@pytest.mark.parametrize('f_name_n_args', array_special_func_map.items())
+def test_cupy_torch_jax_support(xp, data, f_name_n_args):
+    f_name, n_args = f_name_n_args
+    f = getattr(special, f_name)
+    dtypes = ['float32', 'float64']  # good enough for me
+    mbs = npst.mutually_broadcastable_shapes(num_shapes=n_args)
+
+    shapes, final_shape = data.draw(mbs)
+    dtype = data.draw(strategies.sampled_from(dtypes))
+    dtype = getattr(np, dtype)
+    args_np = [np.asarray(data.draw(npst.arrays(dtype, shape)))
+               for shape in shapes]
```

Comment:
@Zac-HD Can the `arrays` strategy generate multiple arrays at a time or just a single array? The examples suggest just one, and I get an error when I pass an iterable or `mutually_broadcastable_shapes` strategy, e.g.:
```python3
npst.arrays(np.float32, npst.mutually_broadcastable_shapes(num_shapes=2)).example()
# hypothesis.errors.InvalidArgument: Array shape must be integer in each dimension, provided shape was ((1,), (2,))
npst.arrays(np.float32, [(1,), (2,)]).example()
# hypothesis.errors.InvalidArgument: Array shape must be integer in each dimension, provided shape was (((), ()), ())

Reply:
Just one at a time - once we started looking at the possibility of per-array dtype and element specification it was just too complicated to put into a single function.  You'll probably want to define an `@st.composite` strategy to generate all your arrays together, or possibly use `st.data()` and draw them interactively inside your test function.  

Re; `provided shape was (((), ()), ())`: that's because `mutually_broadcastable_shapes(...)` generates `((in_shape_1, ..., in_shape_N), output_shape)` so that you can check the actual against expected shape for the signature, but you're treating it as `(not_really_in_shape_1, not_really_in_shape_2)` and the tuple-of-input-shapes is not a valid shape.

Reply:
> or possibly use st.data() and draw them interactively inside your test function.

That's what I'm doing in the code.

> mutually_broadcastable_shapes(...) generates ((in_shape_1, ..., in_shape_N), output_shape) so that you can check the actual against expected

Yup, in the code here `shapes` is the in shapes only.

> but you're treating it as...

I'm not sure I follow. I think the code of the PR is doing the right thing, and I understand now that code in my comment is not supposed to work.

Reply:
Oops, yes, the error messages were swapped. Sorry about that! I'll go ahead and mark this as resolved.